src/components/ErrorMessage.tsx

```diff
@@ -0,0 +1,70 @@
+import { ethersErrorCodes, mobileAndUnder } from "@/constants";
+import type { ErrorMessage as ErrorMessageType } from "@/types";
+import NextLink from "next/link";
+import styled from "styled-components";
+
+export function ErrorMessage({ text, link }: ErrorMessageType) {
+  const isEthersError = ethersErrorCodes.some((code) => text.includes(code));
+
+  function ethersErrorLink() {
+    if (!isEthersError) return null;
+
+    const [firstPart, secondPart] = text.split("[");
+
+    const ethersErrorLink = secondPart
```

Comment:
this kind of ethers specific stuff should be in a utility and processed before this component i would think, so the component could just focus on a single type of data, either that or make 2 error components to handle different data

Reply:
yeah good idea